Auto discovery change for device global

intel-liudean opened the pull request:
For each device global, the init_mode field is changed to can_skip_programming field, a new field reset_on_reuse is added at the end.

zibaiwan commented:
@intel-liudean , thanks for working on this!

I haven't finished the review yet but I think you need to make more updates in the acl_auto_configure_test.cpp. Take a look at the Sophie's change earlier: https://github.com/intel/fpga-runtime-for-opencl/commit/6f68280f5ba865bde96d27ea63f82b411ee3958a, there are many hard-coded device global in several tests, all of them require fields update I believe.

Also please squash those two commits into 1 single commit and provide the description in the commit message and in the PR title. Thanks!

sophimao commented on include/acl.h:
```diff
@@ -514,23 +514,17 @@ typedef enum {
 
 // Enum values here also need to match the SPIRV spec for device
```
Since the enum was removed, I think you can also remove this part of the comment.

intel-liudean commented:
> @intel-liudean , thanks for working on this!
> 
> I haven't finished the review yet but I think you need to make more updates in the acl_auto_configure_test.cpp. Take a look at the Sophie's change earlier: [6f68280](https://github.com/intel/fpga-runtime-for-opencl/commit/6f68280f5ba865bde96d27ea63f82b411ee3958a), there are many hard-coded device global in several tests, all of them require fields update I believe.
> 
> Also please squash those two commits into 1 single commit and provide the description in the commit message and in the PR title. Thanks!

I have addressed your feedback @zibaiwan. Please take a look again.

zibaiwan reviewed (approved):
Looks good to me! Thanks!
